neural_compressor/torch/algorithms/base_algorithm.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+from collections import OrderedDict
+from typing import Any
+
+import torch
+
+algo_quantizers = {}
+
+
+def algo_quantizer_register(name):
+    def decorator(algo_quantizer):
+        algo_quantizers[name] = algo_quantizer
+        return algo_quantizer
+
+    return decorator
+
+
+class Quantizer(ABC):
+    """The base quantizer for one step algorithm quantizers."""
+
+    def __init__(self, config: OrderedDict = {}):
+        self.config = config
+
+    @abstractmethod
+    def quantize(self, model: torch.nn.Module, *args: Any, **kwargs: Any) -> torch.nn.Module:
+        raise NotImplementedError
+
+
+class TwoStepQuantizer(ABC):
+    """The base quantizer for two steps algorithm quantizers."""
+
+    def __init__(self, config: OrderedDict = {}):
+        self.config = config
+
+    @abstractmethod
+    def prepare(self, model: torch.nn.Module, *args: Any, **kwargs: Any) -> torch.nn.Module:
+        raise NotImplementedError
+
+    @abstractmethod
+    def convert(self, model: torch.nn.Module, *args: Any, **kwargs: Any) -> torch.nn.Module:
+        raise NotImplementedError

```

Comment:
Since one-step and two-step use different function names, is it reasonable for us to merge these two quantizers together? @yiliu30

Reply:
The two Quantizer calsses was merged together.